ra/next/resource-agent-api.md

```diff
@@ -398,39 +410,60 @@ Currently, the following additional environment variables are defined:
 
     The name of the resource type being operated on.
 
-#### Monitor-Specific Parameters
+* `OCF_OUTPUT_FORMAT`
+
+   Resource agents may optionally support multiple formats for output generated
+   by an action. The specific formats supported and values used to indicate
+   them are left to agents, but it is recommended to use "text" for readable
+   text output, "xml" for XML output, and "html" for HTML output.
+
+   The `meta-data` action must support, and default to, XML output. The default
+   for other actions is left to agents.
 
-Resource agents may optionally support the parameters listed here when called
-with the `monitor` action.
+#### Check Levels
+
+Resource agents may optionally support the `OCF_CHECK_LEVEL`
+environment variable when called with the `monitor` or `validate-all` actions,
+to allow requests for more or less intensive checks. Agents that support
+`OCF_CHECK_LEVEL` may choose their own default value if it is not specified.
+
+Agents that support `OCF_CHECK_LEVEL` should indicate so by specifying actions
+with `depth` attributes in agent meta-data, as described in
+**Resource Agent Meta-Data**.
```

Comment:
No idea why there's the inconsistency here. Is there any specific/historical reasons? Metadata is the reference for users to know what are supported. It means when users see `depth` is a supported op parameter according to the metadata, but they will need to configure `OCF_CHECK_LEVEL` instead?  It's tricky that users need to somehow manage to know the connection between them ... In case `depth` is configured, should pacemaker translate it into `OCF_CHECK_LEVEL`?

Reply:
Maybe, this is the time to drop `depth=`, and clean up both RA code and this doc?

1. `depth` concept exists quite some years. There are 81 RA advertise it
    resource-agents> grep -r 'monitor" depth="0"' heartbeat/ |sed 's/:.*//'|uniq| wc -l

2. Howver, `OCF_RESKEY_depth` never appears from all commits of resource-agents
    resource-agents> git grep OCF_RESKEY_depth $(git rev-list --all)

3. `OCF_CHECK_LEVEL=` works well alone without `depth=`

Reply:
It's only that it'd look very strange that such a parameter `OCF_CHECK_LEVEL` with all capital letters to be shown in the metadata.

Reply:
I do think the naming difference is purely historical.

depth is just a hint to UIs, so it's unused by the agents themselves or pacemaker (and not passed as an environment variable). It could be used to e.g. automatically generate recurring monitor configs when creating a resource. Pacemaker shouldn't do anything with it because monitors aren't required -- a user could configure just a basic monitor and not use check levels if they want, even when the agent supports it.

It would make sense if the names were consistent, and replacing OCF_CHECK_LEVEL with OCF_RESKEY_depth or OCF_RESKEY_meta_depth seems reasonable. But it would be backward-incompatible, so Pacemaker would have to pass both names to agents, which I think is not worth the trouble.

If you think it's still a good idea for the long run, we could add the new name now and deprecate the old one, requiring cluster managers (Pacemaker) and agents to support both names, and then in a future version drop the old name.

There is a little bit of a time bomb in backward-incompatible changes (which would require changing the major version of the standard). The current standard says that cluster managers and agents must advertise the major version of the standard they support, and refuse to work with other versions. It would be better for upgrades if multiple versions could be supported. The cluster manager could detect the major version supported by the agent from meta-data, and call it appropriately. On the agent end, perhaps the cluster manager could pass the versions it supports via an environment variable, and the agent could work as long as it supported any of them. But I really want to get this 1.1 standard out, so I'd save that for a new proposal.